src/app/filters/filter-fields.component.html

```diff
@@ -34,11 +35,38 @@
           <li *ngFor="let query of currentField.queries"
               [ngClass]="{'selected': query.value === currentValue}">
             <a role="menuitem" tabindex="-1" (click)="selectValue(query)">
+              <span *ngIf="query?.iconClass" class="fa {{query?.iconClass}}"></span>
+              <img class="avatar" *ngIf="query?.imageUrl" [attr.src]="query?.imageUrl" />
               {{query.value}}
             </a>
           </li>
         </ul>
       </div>
     </div>
+    <div *ngIf="currentField.type === 'typeahead'">
```

Comment:
When we discussed type-ahead functionality, I had envisioned updating the placeholder text Vs adding another dropdown. Using an input for this case is fine, but don't believe using a filtered list is the best approach for this component.